### PR TITLE
DEV-840 Wrap up two single sample and somatic in one pipeline

### PR DESCRIPTION
**Pull request by pauldwolfe**
Adds a FullPipeline which will run the two single samples in parallel
and then the somatic.

The sample ids are inferred from a set id passed, so currently only
works with samples in the CPCT12345678R/T format.

### CODE REVIEW

File: cluster/src/main/java/com/hartwig/pipeline/FullPipeline.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        this.somaticPipeline = somaticPipeline;
+        this.executorService = executorService;
+    }
+
+    public PipelineState run() {
+        Future<PipelineState> referenceStateFuture = executorService.submit(referencePipeline::run);
+        Future<PipelineState> tumorStateFuture = executorService.submit(tumorPipeline::run);
+        try {
+            PipelineState singleSampleState =
+                    new PipelineState().combineWith(referenceStateFuture.get()).combineWith(tumorStateFuture.get());
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/FullPipeline.java]: Is there are reason you can't do just `referenceStateFuture.get().combineWith(tumorStateFuture.get());`, without the initial `new` object?

File: cluster/src/test/java/com/hartwig/pipeline/FullPipelineTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        when(somatic.run()).thenReturn(failed());
+        assertThat(victim.run().status()).isEqualTo(PipelineStatus.FAILED);
+    }
+
+    private static PipelineState succeeded() {
+        return new PipelineState();
+    }
+
+    private static PipelineState failed() {
+        PipelineState failedState = succeeded();
```
Comment (nedleitch) [inline cluster/src/test/java/com/hartwig/pipeline/FullPipelineTest.java]: Not sure the few characters saved there are worth the confusion...

Comment (pauldwolfe) [reply cluster/src/test/java/com/hartwig/pipeline/FullPipelineTest.java]: Ha Intellij's decision, will change

File: cluster/src/main/java/com/hartwig/pipeline/FullPipeline.java (hunk 1 above)
Comment (pauldwolfe) [reply cluster/src/main/java/com/hartwig/pipeline/FullPipeline.java]: we could but thought it was a bit arbitrary to use either the reference or tumor... will think about it

**Pull request merged**